src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -798,6 +798,11 @@ def aks_create(cmd,     # pylint: disable=too-many-locals,too-many-statements,to
 
     outbound_type = _set_outbound_type(outbound_type, network_plugin, load_balancer_sku, load_balancer_profile)
 
+    # Validate network plugin to be either "azure" or "kubenet"
+    if network_plugin:
+        if network_plugin not in ('azure', 'kubenet'):
```

Comment:
lower()?

Reply:
We only take lower case "azure" and "kubenet", it is case sensitive if you mean network_plugin should be convert to lower case.